### PR TITLE
Follow-up to Threadsafe randomgen: purge nSpaceSamples and nVelocitySamples

### PR DESCRIPTION
**Pull request by markusbattarbee**

### CODE REVIEW

File: projects/Larmor/Larmor.cpp [hunk 1]
```diff
@@ -108,27 +104,8 @@ namespace projects {
...
-                     for (uint vk=0; vk<this->nVelocitySamples; ++vk)
-                     {
-                        avg += getDistribValue(x+i*d_x, y+j*d_y, z+k*d_z, vx+vi*d_vx, vy+vj*d_vy, vz+vk*d_vz, popID);
-                     }
-      
-      creal result = avg *this->DENSITY * pow(mass / (2.0 * M_PI * kb * this->TEMPERATURE), 1.5) /
-                     (this->nSpaceSamples*this->nSpaceSamples*this->nSpaceSamples) / 
-                     (this->nVelocitySamples*this->nVelocitySamples*this->nVelocitySamples);
+      Real avg = getDistribValue(x+0.5*dx, y+0.5*dy, z+0.5*dz, vx+0.5*dvx, vy+0.5*dvy, vz+0.5*dvz, popID);
+      creal result = avg *this->DENSITY * pow(mass / (2.0 * M_PI * kb * this->TEMPERATURE), 1.5);
```
Comment (ykempf) [inline projects/Larmor/Larmor.cpp:108]: could be one line? :)

File: projects/Shock/Shock.cpp [hunk 2]
```diff
@@ -111,29 +107,8 @@ namespace projects {
...
-         avg += getDistribValue(x+i*d_x, y+j*d_y, z+k*d_z, vx+vi*d_vx, vy+vj*d_vy, vz+vk*d_vz, popID);
-         }
-      
-      creal result = avg *this->DENSITY * pow(mass / (2.0 * M_PI * kb * this->TEMPERATURE), 1.5) /
-                     (this->nSpaceSamples*this->nSpaceSamples*this->nSpaceSamples) / 
-   //                (Parameters::vzmax - Parameters::vzmin) / 
-                     (this->nVelocitySamples*this->nVelocitySamples*this->nVelocitySamples);
-               
+      Real avg = getDistribValue(x+0.5*dx, y+0.5*dy, z+0.5*dz, vx+0.5*dvx, vy+0.5*dvy, vz+0.5*dvz, popID);
+      creal result = avg *this->DENSITY * pow(mass / (2.0 * M_PI * kb * this->TEMPERATURE), 1.5);
```
Comment (ykempf) [inline projects/Shock/Shock.cpp:111]: Here too

File: sysboundary/copysphere.cpp [hunk 3]
```diff
@@ -774,33 +772,13 @@ namespace SBC {
...
-               if (average !=0.0 ) {
-                  data[blockLID*WID3+cellIndex(ic,jc,kc)] = average;
-               }
+               average = shiftedMaxwellianDistribution(
+                  popID,
+                  vxCell + 0.5*dvxCell,
+                  vyCell + 0.5*dvyCell,
+                  vzCell + 0.5*dvzCell
+                  );
+               data[blockLID*WID3+cellIndex(ic,jc,kc)] = average;
```
Comment (ykempf) [inline sysboundary/copysphere.cpp:781]: Here too

File: sysboundary/ionosphere.cpp [hunk 4]
```diff
@@ -3369,37 +3367,15 @@ namespace SBC {
...
-               }
+               average = shiftedMaxwellianDistribution(
+                  popID,
+                  sP.rho,
+                  sP.T,
+                  vxCell + 0.5*dvxCell,
+                  vyCell + 0.5*dvyCell,
+                  vzCell + 0.5*dvzCell
+                  );
+               data[blockLID*WID3+cellIndex(ic,jc,kc)] = average;
```
Comment (ykempf) [inline sysboundary/ionosphere.cpp:3378]: And one more.

File: sysboundary/setmaxwellian.cpp [hunk 5]
```diff
@@ -248,37 +245,15 @@ namespace SBC {
...
-               if (average != 0.0) {
+               average =   maxwellianDistribution(
+                  popID,
+                  rho,
+                  T,
+                  vxCell + 0.5*dvxCell - Vx,
+                  vyCell + 0.5*dvyCell - Vy,
+                  vzCell + 0.5*dvzCell - Vz
+                  );
                   data[blockLID*WID3+cellIndex(ic,jc,kc)] = average;
```
Comment (ykempf) [inline sysboundary/setmaxwellian.cpp:256]: laalaalaa

**Review by alhom — APPROVED** [review]

**Pull request merged**